src/apps/jsgeneric/jsgeneric.cpp

```diff
@@ -64,24 +73,45 @@ namespace ccfapp
           args);
         */
 
-        const auto err_it = response.find(jsonrpc::ERR);
+        auto err_it = response.find("error");
         if (err_it == response.end())
         {
-          const auto result_it = response.find(jsonrpc::RESULT);
+          auto result_it = response.find("result");
           if (result_it == response.end())
           {
-            // Response contains neither RESULT nor ERR. It may not even be an
+            // Response contains neither result nor error. It may not even be an
             // object. We assume the entire response is a successful result.
-            return make_pair(true, response);
+            args.rpc_ctx->set_response_result(std::move(response));
+            return;
           }
           else
           {
-            return make_pair(true, *result_it);
+            args.rpc_ctx->set_response_result(std::move(*result_it));
+            return;
           }
         }
         else
         {
-          return make_pair(false, *err_it);
+          int err_code = jsonrpc::CCFErrorCodes::SCRIPT_ERROR;
+          std::string msg = "";
+
+          if (err_it->is_object())
+          {
+            auto err_code_it = err_it->find("code");
+            if (err_code_it != err_it->end())
+            {
+              err_code = *err_code_it;
+            }
+
+            auto err_message_it = err_it->find("message");
+            if (err_message_it != err_it->end())
+            {
+              msg = *err_message_it;
+            }
+          }
+
+          args.rpc_ctx->set_response_error(err_code, msg);
+          return;
         }
```

Comment:
To clarify the whole section above: The Lua and JS handlers currently need to give a return value and indicate whether this was a success or an error message. They should return an object, either `{result = ...}` or `{error = ...}`, we use the name of the field to work out whether it was an error, and use the `...` as the corresponding payload. In future, we may want to expose the context itself to the interpreter, so the handler can parse the request then set response headers and payload manually.